src/middlewared/middlewared/plugins/directoryservices_/cache.py

```diff
@@ -156,7 +159,10 @@ def _retrieve(self, data, options):
                     if entry is None:
                         return None
 
-                self._insert(data['idtype'], entry)
+                try:
+                    self._insert(data['idtype'], entry)
+                except Exception:
+                    self.logger.warning('Failed to insert %s into directory services cache', entry)
```

Comment:
Should you do self.logger.exception here? Also will KeyError ever occur with these changes since you're catching everything?

Reply:
I added comments explaining the indented exception handling here as well as the origin of KeyError that we're concerned about.